### ISSUE TITLE
[Question] Change locale on the fly

### ISSUE THREAD

**Issue by felpel** [DevExpress/DevExtreme-Intl#49]
Hi,

I'm trying to change the locale of my application at runtime with a language picker like this one:

![image](https://user-images.githubusercontent.com/5000004/38840522-23cc2e58-41ae-11e8-8684-cb4c18beb80d.png)

In the current screenshot, "French" would be the selected language, but I could pick also "English".

It works fine when I call the locale method in my custom localization module after loading my french and english translations as:

```ts
// DevExtreme localization
import { loadMessages, locale } from "devextreme/localization";
import * as enMessages from "devextreme/localization/messages/en.json";
import * as frMessages from "devextreme/localization/messages/fr.json";

loadMessages(frMessages);
loadMessages(enMessages);

locale(navigator.language);
```

If I hardcode `fr` or `en` as the argument for the locale method, I see your translations showing up. Although, after I've selected another language from my combo box, it's seems that recalling `locale(selectedLanguage)` does nothing. **Do I need to refresh the webpage?** It seems that it's the case with your Angular [example](https://js.devexpress.com/Demos/WidgetsGallery/Demo/Localization/UsingIntl/Angular/Light/)...

**Comment by nightskylark** [comment]
Hi,
Our widgets load localized strings only once on initializing. When you need to change your language, it's necessary to reload the current page using `window.location.reload();`.

**Comment by TobiasBreuer** [comment]
@nightskylark Hi, I know this issue is already closed, but is this seriously the final solution for this issue? Especially within an SPA reloading the application for changing the language is not an option.
We are currently using ngx-translate to allow for internationalization of our Angular based SPA. Now the only part that's left are the dx components. I would not have imagined that this would turn out to be a stumble stone at the end.
Is there a plan to change this in the near future, or will this behavior stay as is?
Best regards.

**Comment by nightskylark** [comment]
Hi, @TobiasBreuer 
You mentioned that reloading the window is not an option. Usually, users change a locale only once and therefore your app should be reloaded only once. Moreover, real-world apps should be able to restore its state after reloading. In this case, it is not quite clear why reloading an app for applying the required locale is not an option for you. Would you please describe why the current solution does not meet your requirements? We will consider your scenario during our future developing.

**Comment by lemonway** [comment]
hi, I've got the same problem. 

of course end-users only do it one or two, so we can live with windows.reload() => but it still kind of akward because: a **SPA** must to reload the page entierly while (in plus) it didn't need anything from the server!

- all the dictionary was already loaded on frontend (-> why did we have to reload the pages?)
- reload the page == multiple round-trip to the server (in order to reload resources + mount others component again) == slow experience for end-user

**Comment by duongphuhiep** [comment]
hi, I've got the same problem.

of course end-users only do it one or two, so we can live with windows.reload() => but it still kind of akward because: a SPA must to reload the page entierly while (in plus) it didn't need anything from the server!

- all the dictionary was already loaded on frontend (-> why did we have to reload the pages?)
- reload the page == multiple round-trip to the server (in order to reload resources + mount others component again) == slow experience for end-user

**Comment by ChristofFritz** [comment]
To bring the problem to a point: In the world of Angular where we have observables it's simply just stupid to do a full page reload just for a couple of strings. Even though @nightskylark is absolutely right with his argmentation that the users won't change the locale all the time.
It would be a "really nice to have" if the locale changes on the fly.

**Comment by m--c** [comment]
Hi there,
Just wanted to say that this is the only library I saw i think in the entire SPA world where you have to actually reload the entire app by design. I know its not the end of the world, its just weird that you present this as a perfectly normal, fine, final solution. It's also usually one of the first interactions the user has with the app and also one of the first things developers face when stating an new app with this framework
